include/wil/registry_basic.h

```diff
@@ -962,28 +954,86 @@ namespace reg_view_details
                     continue;
                 }
 
-                return err_policy::HResult(HRESULT_FROM_WIN32(error));
+                return get_err_policy::HResult(HRESULT_FROM_WIN32(error));
             }
 
             ::wil::registry::reg_view_details::trim_buffer(return_value);
-            return err_policy::HResult(S_OK);
+            return get_err_policy::HResult(S_OK);
+        }
+
+    public:
+
+        template <typename R>
+        typename err_policy::result get_value(_In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, R& return_value, DWORD type = ::wil::registry::reg_view_details::get_value_type<R>()) const
+        {
+            return get_value_internal(subkey, value_name, return_value, type);
         }
 
         template <typename R>
-        typename err_policy::result get_value(_In_opt_ PCWSTR value_name, optional_value<R>& return_value, DWORD type = ::wil::registry::reg_view_details::get_value_type<R>()) const
+        typename err_policy::result get_value(_In_opt_ PCWSTR value_name, R& return_value, DWORD type = ::wil::registry::reg_view_details::get_value_type<R>()) const
         {
             return get_value(nullptr, value_name, return_value, type);
         }
 
         // intended for err_exception_policy as err_returncode_policy will not get an error code

```

Comment:
Should we use enable_ifs for these functions? And below.

Reply:
@citelao , would that give better compiler errors than what would be provided already? I'm not really sure :(